common/src/main/java/org/apache/uniffle/common/netty/protocol/GetLocalShuffleDataResponse.java

```diff
@@ -18,53 +18,64 @@
 package org.apache.uniffle.common.netty.protocol;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
 
+import org.apache.uniffle.common.netty.buffer.FileSegmentManagedBuffer;
+import org.apache.uniffle.common.netty.buffer.ManagedBuffer;
+import org.apache.uniffle.common.netty.buffer.NettyManagedBuffer;
 import org.apache.uniffle.common.rpc.StatusCode;
 import org.apache.uniffle.common.util.ByteBufUtils;
 
-public class GetLocalShuffleDataResponse extends RpcResponse {
-  private ByteBuf data;
+public class GetLocalShuffleDataResponse extends RpcResponse implements Transferable {
 
-  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, byte[] data) {
-    this(requestId, statusCode, null, data);
-  }
-
-  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, String retMessage, byte[] data) {
-    this(requestId, statusCode, retMessage, Unpooled.wrappedBuffer(data));
-  }
+  private ManagedBuffer buffer;
 
-  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, String retMessage, ByteBuf data) {
+  public GetLocalShuffleDataResponse(long requestId, StatusCode statusCode, String retMessage, ManagedBuffer data) {
     super(requestId, statusCode, retMessage);
-    this.data = data;
+    this.buffer = data;
   }
 
   @Override
   public int encodedLength() {
-    return super.encodedLength() + Integer.BYTES + data.readableBytes();
+    return super.encodedLength() + Integer.BYTES + buffer.size();
   }
 
   @Override
   public void encode(ByteBuf buf) {
     super.encode(buf);
-    ByteBufUtils.copyByteBuf(data, buf);
-    data.release();
+    if (buffer instanceof FileSegmentManagedBuffer) {
+      buf.writeInt(buffer.size());
```

Comment:
Why do we only write the size of buffer?

Reply:
How do the other systems like Spark, Celeborn implement this?

Reply:
In order to keep the uniffle interface design as much as possible, reference Celeborn add ManagedBuffer to hold different type ShuffleData(bytebuf or FileRegion).

Reply:
> In order to keep the uniffle interface design as much as possible, reference Celeborn&Spark add ManagedBuffer to hold different type ShuffleData(bytebuf or FileRegion).

Ok for me.